src/GraphQL.Federation/readme.md

```diff
@@ -0,0 +1,39 @@
+## Installation
+
+You can install the latest stable version via [NuGet](https://www.nuget.org/packages/GraphQL.Federation/).
+```
+> dotnet add package GraphQL.Federation
+```
+
+## Configuration
+
+Register `FederatedInstrumentFieldMiddleware` in the DI
+```
+services.AddSingleton<FederatedInstrumentFieldMiddleware>();
+```
+Configure document executor and collect trace
+```
+var result = await _executer.ExecuteAsync(options =>
+{
+    // check if the tracing is enabled through header
+    if (context.Request.IsFederatedTracingEnabled())
```

Comment:
Should be property in options.

Reply:
Federated tracing is controlled through  http header. `IsFederatedTracingEnabled` returns true if the header is set. I am using it to determine if i should enable metrics and use field middleware. Could you please elaborate you idea?

Reply:
Tracing (federated or not) is just an auxilary data structure aggregated when execution fields. This code can be performed (or not performed) regardless of specific transport (HTTP in your case). So let's use simple option (boolean property in options). And of course this option can be set from HTTP headers but **in some other place** (transport-specific library or application).

Reply:
Oh got it. I was going to move HttpRequestExtensions out of GraphQL.NET. I will update the readme once i do that.  I will put PR in graphql-dotnet/server